### ISSUE TITLE
GRC Client/Wallet closing in background.

### ISSUE THREAD

**Issue by mpenguin** [gridcoin-community/Gridcoin-Research#712]
Win10 x64, 16gb RAM, fully updated. No other hardware/software issues.

My GRC wallet opens fine and will sync up. I sent a beacon about 10 days ago (23 October 2017) and am still receiving POR payments. However, it will close quietly after an hour or two. It doesn't matter if I leave it open on the desktop in focus or in the background or if I close it to the taskbar. It always closes.

I've check the config file and everything looks in order. I've uninstalled and reinstalled (backed up Wallet and Config, then replaced) and it's still happening.

Is there someway I can run a debug log that I can upload and share? Thanks.

**Comment by tomasbrod** [comment]
This issue is known one. You can help a little by uploading your `debug.log` file.
**Warning: the file is overwritten every time you restart the wallet.**
Locate your gridcoin folder (it is in `%appdata%`) and drop it here as a comment ("Attach files by dragging & dropping, selecting them, or pasting from the clipboard.").

**Comment by mpenguin** [comment]
This is the file from the last time it closed unexpectedly. Thanks!

[debug.log](https://github.com/gridcoin/Gridcoin-Research/files/1437467/debug.log)

**Comment by iFoggz** [comment]
hey there, looking at your debug.log it looks like you sent the debug from after the wallet was restarted. however the logs are cleared when a new session starts. Also I'm seeing in the logs you have multiple external CPIDs. are all the projects in one boinc client only? you may have a split cpid issue there.

however no information is in your logs regarding crash because you restarted the client so next time the client crashes please send the log before restarting the wallet :)

**Comment by iFoggz** [comment]
also your autounlock is failing.

**Comment by mpenguin** [comment]
Weird. The debug.log was definitely sent before I restarted. I'll send one again later today after the client exits.
As for multiple CPIDs, I'm not sure. I only have one Wallet, but I have BOINC running on 3 machines with various IPs. "list cpids" on the console showed that all my CPIDs for various projects are the same one. I've also fixed the autounlock. Thanks.

**Comment by mpenguin** [comment]
It quit again today while I was out. There's definitely something funky going on at the beginning (end?) of the debug log.
[debug.log](https://github.com/gridcoin/Gridcoin-Research/files/1443861/debug.log)

**Comment by pbiebricher** [comment]
Hi,

my client is also closing silently. From my monitor and restart script:
$ grc_dog.sh
2017-10-30 07:54:45 start grc_dog.sh
2017-11-03 03:58:55 restart gridcoinresearch in 30s
2017-11-06 04:03:20 restart gridcoinresearch in 30s
2017-11-06 10:55:27 restart gridcoinresearch in 30s
2017-11-06 14:51:59 restart gridcoinresearch in 30s
2017-11-06 15:42:43 restart gridcoinresearch in 30s
2017-11-06 19:59:21 restart gridcoinresearch in 30s

I found messages in the windows event log, see file. Perhaps this helps.
[grcclient_events.txt](https://github.com/gridcoin/Gridcoin-Research/files/1450114/grcclient_events.txt)

**Comment by denravonska** [comment]
@pbiebricher That looks very interesting:

```
Ausnahmeinformationen: System.IO.IOException
   bei System.IO.__Error.WinIOError(Int32, System.String)
   bei System.IO.FileStream.Init(System.String, System.IO.FileMode, System.IO.FileAccess, Int32, Boolean, System.IO.FileShare, Int32, System.IO.FileOptions, SECURITY_ATTRIBUTES, System.String, Boolean, Boolean, Boolean)
   bei System.IO.FileStream..ctor(System.String, System.IO.FileMode, System.IO.FileAccess, System.IO.FileShare)
   bei GRCSec.modUpgrade.GetMd5OfFile(System.String)
   bei GRCSec.GridcoinData.ExtractProjectsFromMasterDatabaseFile(System.String, System.String ByRef, System.String ByRef)
   bei GRCSec.GridcoinData.CastVotes()
   bei GRCSec.GridcoinData._Lambda$__1(System.Object)
   bei System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(System.Object)
   bei System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bei System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bei System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   bei System.Threading.ThreadPoolWorkQueue.Dispatch()
   bei System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

Could that finally be the exception we have been chasing?

**Comment by SCCMAdmin** [comment]
Hi All, I too have had this issue in the past. Not recently as I haven't opened the wallet lately. Let me know if you need help with this still. Cheers.

**Comment by moisespr123** [comment]
Crashes still happening in v3.6.3.0 while listen=1 on Windows 10.

**Comment by SCCMAdmin** [comment]
Happy New Year!

Same here gang. v3-6-3-0-unk on Win10 1709 x64 everything else working fine. After some hours or some nights the wallet has silently exited the building! :-)  Need to relaunch it manually. I wonder if there is some auto restart option after a crash?  It could help hide crashes if used at the wrong time, but when experiencing crashes and waiting for a fix, now would be the best time for such a switch.

**Comment by denravonska** [comment]
Should be fixed in 3.7.0.0.
